serverless/src/main/java/com/iluwatar/serverless/baas/model/Person.java

```diff
@@ -84,23 +91,55 @@ public void setAddress(Address address) {
 
   @Override
   public boolean equals(Object o) {
+    boolean nullObj = false;
+    coverage[8] = true; //Default
     if (this == o) {
+      coverage[0] = true;
       return true;
     }
+
+    if (o == null) {
+      nullObj = true;
+      coverage[1] = true;
+    }
+
+    if (nullObj) {
+      return false;
+    }
+
     if (o == null || getClass() != o.getClass()) {
+      if (getClass() != o.getClass()) {
+        coverage[2] = true;
+      }
       return false;
     }
 
+
     Person person = (Person) o;
 
     if (firstName != null ? !firstName.equals(person.firstName) : person.firstName != null) {
+      if (firstName != null && !firstName.equals(person.firstName)) {
+        coverage[3] = true;
+      }
+      if (firstName == null && person.firstName != null) {
+        coverage[4] = true;
+      }
       return false;
     }
 
     if (lastName != null ? !lastName.equals(person.lastName) : person.lastName != null) {
+      if (lastName != null && !lastName.equals(person.lastName)) {
+        coverage[5] = true;
+      }
+      if (lastName == null && person.lastName != null) {
+        coverage[6] = true;
+      }
       return false;
     }
 
+    if (address != null) {
+      coverage[7] = true;
+    }
     return address != null ? address.equals(person.address) : person.address == null;
   }
```

Comment:
Need to check address.equals(person.address) and person.address == null as well

Reply:
Fixed! :)